core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -620,6 +621,10 @@ private synchronized void updateSplitQueueSpace()
     private synchronized void processTaskUpdate(TaskInfo newValue, List<SplitAssignment> splitAssignments)
     {
         updateTaskInfo(newValue);
+        if (taskStatusFetcher.getTaskStatus().getState() == INITIALIZING) {
+            pendingRequestsCounter.incrementAndGet();
```

Comment:
you shouldn't update when `pendingSourceSplitCount == 0`, should you?

BTW: why `if (taskStatusFetcher.getTaskStatus().getState() == INITIALIZING)` is needed? It seems it would work without it. Also, ignoring `splitAssignments` looks wrong. Assertion (that it's empty) would be useful at least

Reply:
the task isn't running, and coordinator shouldn't treat it as running? Maybe that doesn't matter?

I felt continuing to assign splits to a task that is still in init stage didn't make sense, because they won't get processes.